detekt-compiler-plugin-gradle-plugin/src/main/kotlin/io/github/detekt/gradle/DetektKotlinCompilerPlugin.kt

```diff
@@ -74,15 +73,15 @@ class DetektKotlinCompilerPlugin : KotlinCompilerPluginSupportPlugin {
         }
 
         val options = project.objects.listProperty(SubpluginOption::class.java).apply {
-            add(SubpluginOption(Options.debug, taskExtension.debug.get().toString()))
-            add(SubpluginOption(Options.configDigest, taskExtension.config.toDigest()))
-            add(SubpluginOption(Options.isEnabled, taskExtension.isEnabled.get().toString()))
-            add(SubpluginOption(Options.useDefaultConfig, taskExtension.buildUponDefaultConfig.get().toString()))
-            add(SubpluginOption(Options.allRules, taskExtension.allRules.get().toString()))
-            add(SubpluginOption(Options.disableDefaultRuleSets, taskExtension.disableDefaultRuleSets.get().toString()))
-            add(SubpluginOption(Options.parallel, taskExtension.parallel.get().toString()))
-            add(SubpluginOption(Options.rootPath, project.rootDir.toString()))
-            add(SubpluginOption(Options.excludes, taskExtension.excludes.get().encodeToBase64()))
+            add(SubpluginOption("debug", taskExtension.debug.get().toString()))
```

Comment:
Going back to untyped strings isn't ideal, but to keep them means adding another module that both the Gradle plugin and the compiler plugin depend on.

The Kotlin compiler will error out if an unrecognised option is passed, so spelling errors here will be picked up for the most part. In future, functional tests should also exercise all options.